### PR TITLE
submitting the code

### PR DESCRIPTION
**Pull request by mrajancsr**

### CODE REVIEW

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+from itertools import combinations
+
+class PriorityQueue:
+    """Implementation of a priority queue using Heaps"""
+    def __init__(self):
+        self._queue = []
+        self._index = 0 
+    
+    def push(self, item, priority):
+        """pushes an item with a given priority
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:19]: I am not familiar with this style of docstring?

Comment (Jackal08) [reply Open-Source-Soldier-of-Fortune/raj_submission/mst.py:19]: Missing grammar...
* Periods
* Start a sentence with a capital letter.

Comment (Jackal08) [reply Open-Source-Soldier-of-Fortune/raj_submission/mst.py:19]: These docstrings show that you used a different IDE to Pycharm? They way that you expressed the params, they won't show up in the sphinx documentation.

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+Author: Rajan Subramanian
+Date: 07/16/2020
+"""
+
+import numpy as np 
+import yfinance as yf
+import heapq
+from itertools import combinations
+
+class PriorityQueue:
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:12]: I do like that you are showing command of algorithms and data structures.

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        return heapq.heappop(self._queue)
+    
+    def __len__(self):
+        return len(self._queue)
+    
+    def is_empty(self):
+        return len(self) == 0
+
+class Graph:
+    """implementation of a graph using adjacency map
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:43]: The grammar is a sore point for me.

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+    For a pair of vertices (u,v), (u,z) that has an edges E, F
+    is represented by {u: {v: E, z: F}}"""
+
+    """nested Vertex and Edge classes"""
+    class __Vertex:
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:48]: Noting the use of double underscores for private methods. I rarely see this done. I like that you used it, although our convention is to use the single underscore.

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+            return self._value 
+        
+        def __hash__(self):
+            """allows vertex to be a key in a dictionary"""
+            return hash(id(self))
+        
+        def __repr__(self):
+            return """Vertex({!r})""".format(self._value)
+    
+    # --  nested edge class
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:66]: Why `--`?

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+    class __Edge:
+        """Implements the edge structure that returns edge associated
+            with vertex (u,v)
+        """
+        # light weight edge structure
+        __slots__ = '_start', '_end', '_value'
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:72]: This is new to me, what does __slots__ do? It is a tuple?

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py (hunk 5 above)
Comment (Jackal08) [reply Open-Source-Soldier-of-Fortune/raj_submission/mst.py:66]: To show nesting?

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+        
+        def _calculate_correlation(self, prices):
+            """calculates the correlation given prices
+            Args: 
+            prices: pandas dataframe
+
+            Returns: 
+            correlation:  pandas dataframe
+            """
+            return np.log(1 + prices.pct_change()).corr()
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:261]: Ah ok. Sure we can do log(1 + r)

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+        import networkx as nx
+        import matplotlib.pyplot as plt 
+        g = nx.Graph()
+        # get the edge vertices and edge weights from mst_tree
+        items = (e.get_items() for e in mst_tree)
+        # add the edges in the networkx graph for plotting
+        g.add_weighted_edges_from(items)
+        nx.draw(g, with_labels=True);
+        plt.draw();
+
```
Comment (Jackal08) [inline Open-Source-Soldier-of-Fortune/raj_submission/mst.py:344]: Remove all these white spaces. What IDE are you using? Your code would benefit by adding a linter.

Comment (mrajancsr) [reply Open-Source-Soldier-of-Fortune/raj_submission/mst.py:344]: Hey, I just saw all this.  I'm using vscode.  I just installed that few weeks ago and hence have been using that.   Still trying to my way around the ide.

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py (hunk 6 above)
Comment (mrajancsr) [reply Open-Source-Soldier-of-Fortune/raj_submission/mst.py:72]: It's a dictionary for streamlined memory usage in python.

File: Open-Source-Soldier-of-Fortune/raj_submission/mst.py (hunk 5 above)
Comment (mrajancsr) [reply Open-Source-Soldier-of-Fortune/raj_submission/mst.py:66]: > Why `--`?

Made it easier for me to view the vertex in a list in ipython.

**Pull request merged**